pkg/tools/samples/kustomize.yaml

```diff
@@ -0,0 +1,11 @@
+- name: kustomize
+  description: "A tool to customize Kubernetes resource configurations. Use it to render and apply declarative configurations from a directory containing a kustomization.yaml file."
+  command: "kustomize"
+  command_desc: |
+    The kustomize command-line interface.
+    
+    Core subcommands and usage patterns:
+    - `kustomize build <kustomization_dir>`: Prints the customized resources to standard output. This is useful for inspecting the final configuration before applying it.
+    - `kustomize build <kustomization_dir> | kubectl apply -f -`: A common pattern to apply the output directly to the cluster.
+    
+    Note: `kubectl apply -k <dir>` is a shorthand for the pipe command above and is often preferred.
```

Comment:
I'd say for quick, local development, `kubectl apply -k` is convenient and works perfectly well.

However, for production environments, CI/CD pipelines, and team collaboration, the best practice is to use `kustomize build <dir> | kubectl apply -f -`. It provides superior reproducibility, access to the latest features, and the critical ability to inspect and audit your generated configurations before they are applied.